src/Notepads/Controls/GoTo/GoToControl.xaml.cs

```diff
@@ -15,12 +17,16 @@ public sealed partial class GoToControl : UserControl
 
         public event EventHandler<GoToEventArgs> OnGoToButtonClicked;
 
+        public event EventHandler<KeyRoutedEventArgs> OnGoToControlKeyDown;
+
         private int _currentLine;
 
         private int _maxLine;
 
         private readonly ResourceLoader _resourceLoader = ResourceLoader.GetForCurrentView();
 
+        private readonly KeyboardShortcut<bool> _nativeKeyboardShortcut = new KeyboardShortcut<bool>(false, false, false, VirtualKey.Escape, null);
```

Comment:
I don't get this, can you explain?

Reply:
> I don't get this, can you explain?

The `_native KeyboardShortcuts`/`_nativekeyboardShortcut` variables store all the keyboard shortcut of FindReplace/GoTo control respectively. In the event of key down, the keyboard shortcut is compared with it and if the provided shortcut is not implemented then the shortcut is passed to the editor which then handles it.
The reason I didn't implement KeyboardHandler is that it will only work when the control is on focus. Hence it is better to have shortcuts handled in xaml and editor shortcuts passed via the keydown event.

Reply:
Sorry, I mean this Esc thingy.

Reply:
You are right, I added this to handle the Esc key action in the control itself. But since it is already handled in editor we can remove it.